### DISCUSSION TITLE
"Priority" header

### DISCUSSION THREAD

**Discussion by rfc2822** [bitfireAT/webdav-push#33] [General]
Push service on mobile devices (like FCM) have normal vs high priority pushes:

- Normal pushes are used for normal background data transfer. Device goes into doze after a few seconds again, app stays in the background and can enqueue the actual data transfer.
- High-priority pushes are meant for user interaction. So the device wakes up and can immediately do things. This is usually restricted by some quota.

It's probably a good idea to allow servers to specify high-priority pushes for changes in events in the near future (maybe 15 minutes)?

---

On DAVx5 side, we may consider showing a (low-priority) "pending changes" notification when a sync has been enqueued by a push message, but not been run yet. Users could then tap the notification to open DAVx5 and run the sync. When the sync has run (either automatically or initiated by the user), the notification would disappear again.

**Comment by devvv4ever** [comment]
Sounds good with the notification :)

**Comment by nursoda** [comment]
Since GitHub did not automatically add a back-reference here in discussions (as it does in issues), I'll add that manually:
According to @rfc2822, https://github.com/bitfireAT/webdav-push/pull/35 addresses this discussion RFE.
